### PR TITLE
Toc block wrapper

### PR DESCRIPTION
**Pull request by carl-textalk**

### CODE REVIEW

**Review by kalaspuffar — APPROVED** [review]

**Comment by carl-textalk** [comment]
@bertfrees @PaulRambags Please review the following change regarding the wrapping of toc-entries.

**Review by bertfrees — CHANGES_REQUESTED** [review]
`toc-entry` can now only have one attribute,`ref-id`. All the other need to be moved to the `toc-block` wrapper.

Other than that it looks OK, assuming that the "toc-hd" mode only produces inline content. (I quickly checked but note that I don't really know this code enough. DAISY Pipeline doesn't make use of this module.)

**Comment by carl-textalk** [comment]
> `toc-entry` can now only have one attribute,`ref-id`. All the other need to be moved to the `toc-block` wrapper.

I have moved the attributes from toc-entry to toc-block.

**Comment by bertfrees** [comment]
Good.

File: src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl [hunk 1]
```diff
@@ -267,20 +267,28 @@
 	
 	<xsl:template match="dtb:level1|dtb:level2" mode="toc">
 		<xsl:if test="dtb:h1|dtb:h2">
-<!--
-			<xsl:choose>
-				<xsl:when test="self::dtb:level1 and @class='part'"><toc-entry ref-id="{generate-id(.)}" padding-bottom="1" keep="page"><xsl:apply-templates select="dtb:h1" mode="toc-hd"/></toc-entry><xsl:apply-templates mode="toc"/></xsl:when>
-				<xsl:otherwise>--><toc-entry ref-id="{generate-id(dtb:h1|dtb:h2)}" block-indent="{$toc-indent-multiplier}" text-indent="{2*$toc-indent-multiplier}" keep="page"><xsl:apply-templates select="dtb:h1|dtb:h2" mode="toc-hd"/><xsl:apply-templates mode="toc"/></toc-entry><!--</xsl:otherwise>
-			</xsl:choose>-->
+			<toc-block block-indent="{$toc-indent-multiplier}" text-indent="{2*$toc-indent-multiplier}" keep="page">
```
Comment (PaulRambags) [inline src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl:270]: The `toc-block` has no attribute `keep`

File: src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl [hunk 2]
```diff
@@ -267,20 +267,28 @@
...
 		</xsl:if>
 	</xsl:template>
 	
 	<xsl:template match="dtb:level3|dtb:level4|dtb:level5|dtb:level6" mode="toc">
 		<xsl:if test="dtb:h3|dtb:h4|dtb:h5|dtb:h6">
-			<toc-entry ref-id="{generate-id(dtb:h3|dtb:h4|dtb:h5|dtb:h6)}" block-indent="{$toc-indent-multiplier}" text-indent="{$toc-indent-multiplier}" keep="page"><xsl:apply-templates select="dtb:h3|dtb:h4|dtb:h5|dtb:h6" mode="toc-hd"/>
-			<xsl:if test="dtb:level3 and ancestor::dtb:level1[@class='part']"><xsl:apply-templates mode="toc"/></xsl:if>
-			</toc-entry>
-			<xsl:if test="not(dtb:level3 and ancestor::dtb:level1[@class='part'])"><xsl:apply-templates mode="toc"/></xsl:if>
+			<toc-block block-indent="{$toc-indent-multiplier}" text-indent="{$toc-indent-multiplier}" keep="page">
```
Comment (PaulRambags) [inline src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl:281]: The `toc-block` has no `keep` attribute. We decided not to use the special attributes

**Review by PaulRambags — CHANGES_REQUESTED** [review]

File: src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl (hunk 2 above)
Comment (carl-textalk) [reply src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl:281]: Ok, so i should remove the `keep` attribute from from toc-blocks?

Comment (PaulRambags) [reply src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl:281]: @bertfrees where should the `keep` attribute but put? The only element that has a `keep` attribute is `block` but there is no such thing in the `table-of-contents`.

Comment (bertfrees) [reply src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl:281]: > We decided not to use the special attributes

There has been some confusion. I indeed said somewhere in the discussion that `toc-entry` doesn't have the special attributes, but this is only the case in the documentation. In the RelaxNG schema `toc-entry` does have all the attributes actually. When we were discussing the attributes for `toc-block` we also included all the attributes in the discussion.

File: src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl (hunk 1 above)
Comment (PaulRambags) [reply src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl:270]: After much discussion, we have decided that the `toc-block` will have all `block` attributes, that includes attribute `keep`. So I will approve this pull request.

**Review by PaulRambags — APPROVED** [review]

**Comment by bertfrees** [comment]
@carl-textalk Just to be clear, you may always ask my review of course, but you don't need my approval for changes in this repo. I'm only really interested in "dotify.api" and "dotify.formatter.impl" as these are the modules used in DAISY Pipeline. The other stuff (the braille translators in "dotify.translator.impl" and the x-to-obfl transformations in "dotify.task.impl") can also be accessed from Pipeline but I have no code depending on it.

**Pull request merged**